contracts/common/mixin/RootChainValidator.sol

```diff
@@ -1,95 +1,78 @@
-import { Common } from "../lib/Common.sol";
 import { RLPEncode } from "../lib/RLPEncode.sol";
-import { IRootChain } from "../root/IRootChain.sol";
-import { WithdrawManager } from "../root/WithdrawManager.sol";
-import { DepositManager } from "../root/DepositManager.sol";
-
 import { Lockable } from "./Lockable.sol";
+
+import { IRootChain } from "../../root/IRootChain.sol";
+import { IWithdrawManager } from "../../root/withdrawManager/IWithdrawManager.sol";
+import { Registry } from "../Registry.sol";
 import { RootChainable } from "./RootChainable.sol";
 
 
-/**
- * @title RootChainValidator
- */
 contract RootChainValidator is RootChainable, Lockable {
+  using RLPReader for bytes;
+  using RLPReader for RLPReader.RLPItem;
   using Merkle for bytes32;
-  using RLP for bytes;
-  using RLP for RLP.RLPItem;
-  using RLP for RLP.Iterator;
-
-  // managers
-  DepositManager public depositManager;
-  WithdrawManager public withdrawManager;
-
-  // child chain contract
-  address public childChainContract;
+  
+  // using RLP for RLP.Iterator;
+  Registry internal registry;
+  IRootChain internal rootChain;
 
-  // set deposit manager
-  function setDepositManager(address _depositManager) public onlyOwner {
-    require(_depositManager != address(0));
-    depositManager = DepositManager(_depositManager);
-  }
-
-  // set withdraw manager
-  function setWithdrawManager(address _withdrawManager) public onlyOwner {
-    require(_withdrawManager != address(0));
-    withdrawManager = WithdrawManager(_withdrawManager);
-  }
-
-  // set child chain contract
-  function setChildChainContract(address _childChainContract) public onlyOwner {
-    require(_childChainContract != address(0));
-    childChainContract = _childChainContract;
+  constructor(Registry _registry, IRootChain _rootChain)
+    public
+  {
+    registry = _registry;
+    rootChain = _rootChain;
   }
 
   // validate transaction
   function validateTxExistence(
     uint256 headerNumber,
-    bytes headerProof,
+    bytes memory headerProof,
 
     uint256 blockNumber,
     uint256 blockTime,
 
     bytes32 txRoot,
     bytes32 receiptRoot,
-    bytes path,
+    bytes memory path,
 
-    bytes txBytes,
-    bytes txProof
+    bytes memory txBytes,
+    bytes memory txProof
   ) public view returns (bool) {
     // fetch header block
     bytes32 headerRoot;
     uint256 start;
-    (headerRoot, start,,) = IRootChain(rootChain).headerBlock(headerNumber);
+    (headerRoot, start,,) = rootChain.getHeaderBlock(headerNumber);
 
     // check if tx's block is included in header and tx is in block
-    return keccak256(blockNumber, blockTime, txRoot, receiptRoot)
+    return keccak256(abi.encodePacked(blockNumber, blockTime, txRoot, receiptRoot))
       .checkMembership(blockNumber - start, headerRoot, headerProof) &&
     MerklePatriciaProof.verify(txBytes, path, txProof, txRoot);
   }
 
   // validate transaction & receipt
   function validateTxReceiptExistence(
```

Comment:
Let's clean up the functions here. e.g. calling this func `validateReceiptExistence` and this function calling `validateTxExistence` sounds weird.